### PR TITLE
Feature: retry on subset of jobs hosts

### PR DESCRIPTION
**Pull request by AlanCoding**
Connect https://github.com/ansible/awx/issues/219

### CODE REVIEW

**Comment by AlanCoding** [comment]
Additions to make:

 - add "unreachable" status to choices
 - provide host summary for a GET to /relaunch/ endpoint

File: awx/api/serializers.py [hunk 1]
```diff
@@ -2642,9 +2643,21 @@ class Meta:
...
-    class Meta:
-        fields = ('passwords_needed_to_start',)
+    retry_counts = serializers.SerializerMethodField()
+    hosts = serializers.ChoiceField(
+        required=False, allow_null=True, default='all',
+        choices=[
+            ('all', _('No change to job limit')),
+            ('failed', _('All failed and unreachable hosts')),
+            ('ok', _('Hosts with 1 or more ok tasks')),
+            ('changed', _('Hosts with 1 or more changed tasks')),
```
Comment (AlanCoding) [inline awx/api/serializers.py]: @wenottingham are we going to awx "ok", "changed" as options in the API here? (as opposed to just the UI)

Comment (wenottingham) [reply awx/api/serializers.py]: "ok", "failed"

Comment (AlanCoding) [reply awx/api/serializers.py]: "all" is the default option that doesn't change the limit from the prior job. "ok" is a subset that doesn't include skipped hosts. From discussions up to this point, that would imply that the options we're surfacing are "all", "failed", and "unreachable"

**Review by matburt — APPROVED** [review]

**Pull request merged**